### DISCUSSION TITLE
Production I1 AC energy

### DISCUSSION THREAD

**Discussion by Pierret-ha** [WillCodeForCats/solaredge-modbus-multi#689] [General]
Hello, In the Modbus integration for the energy dashboard, the I1 AC energy sensor incorrectly includes battery discharge, leaving no way to track solar production when a battery is present. At night, both the energy consumed from the battery and the energy consumed from solar production are counted. I've attempted various calculations with I1 and B1, but the only reliable method to obtain the solar production value is by summing the output from the optimizers.

**Comment by drumtrich** [comment]
Hi.
I have the same problem. Does it affect everyone who has a battery and uses this integration, or are there configurations with a battery where solar production is measured correctly?

**Comment by PaulEast** [comment]
I've noticed the same, I'm sure this hasn't always been the case but I've only noticed it recently. This evening I'm seeing 1.7kWh consumed battery AND 1.7kWh of solar generation (after the sun has set).
![image](https://github.com/user-attachments/assets/f97d26a6-1c94-42e4-8acd-11aa00e33b20)

**Reply by PaulEast** [reply]
Ignore that, I've gone back and looked and it appear to always have been like this and I've jjust not noticed!

**Reply by Pierret-ha** [reply]
No, it's a hardware inverter problem from solaredge since their update at August last year. They tell me with another update at January, they fix it but nothing, the problem is the same. We should fix that with an entities in HA

**Reply by PaulEast** [reply]
Thank you, I'm looking at using  entities now, just need some sun to test it now :)

**Comment by AchWasss** [comment]
Any updates on this topic? I have the same issue and it kinda drives me nuts...

**Reply by PaulEast** [reply]
I'm not sure about a fix on the inverter (I'll be honest I didn't look into it) but I have set up 2 helpers based on the work here:

https://gist.github.com/Ashpork/f80fb0d3cb22356a12ed24734065061c (I'm sorry I can't remember where I came across this!)

The 2 I have are: 
1) An integral sensor based on the solar generated W, using a left Riemann sum and measured in kWh.
2) A utility meter based on the above sensor, resetting daily.

I'm definately now expert in any of this but finding my way through a little at a time! I hope this helps.
